cypress/integration/channel/message_draft_then_switch_channel_spec.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Message Draft and Switch Channels', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('M14358 Message Draft Pencil Icon Visible in Channel Switcher', () => {
+        // 1. In a test channel, type some text in the message input box
+        // 2. Do not send the post
+        cy.get('#sidebarItem_town-square').scrollIntoView();
+        cy.get('#sidebarItem_town-square').should('be.visible').click();
+
+        // * Validate if the channel has been opened
+        cy.url().should('include', '/ad-1/channels/town-square');
+
+        // * Validate if the draft icon is not visible on the sidebar before making a draft
+        cy.get('#sidebarItem_town-square').scrollIntoView();
+        cy.get('#sidebarItem_town-square #draftIcon').should('be.not.visible');
+
+        // Type in some text into the text area of the opened channel
+        cy.get('#post_textbox').type('whalecome to the jungle');
```

Comment:
😁 
Though not specified, I suggest to change it to something like
```suggestion
        cy.get('#post_textbox').type('message draft test');
```